### ISSUE TITLE
Add sung syllable detection

### ISSUE THREAD

**Issue by SteveALee** [music-practice-tools/relative-notes#8]
I often sing one function syllable / number  whist saying another and an error alert would be useful.

That requires speech to text processing:

- The platform API is only supported by default on Edge and Chrome.
- The platform API uses a remote service, but does allow supplying a grammar to improve accuracy
-  a version of Google's Tensorflow supplied as a JS lib 'speech comands' handles number recognition but singing is very likely to confuse it (pitch shifts, sustained vowels, and vibrato)
-  Use WASM wrapped native recognition engine like Whisper - this should handle singing, Solfege etc

## Detailed Gemini results

Standard models like TensorFlow.js Speech Commands struggle with sung words.

The model converts audio into a visual spectrogram to detect patterns. Singing introduces pitch shifts, sustained vowels, and vibrato, which distort those audio spectrograms and sharply reduce accuracy.

How to fix it:

    1 Use Transfer Learning: Retrain the TensorFlow.js model using its built-in fine-tuning feature. Collect a few dozen samples of your users singing the target digits, and train a custom layer right in the browser.
    2 Switch to Whisper: Larger neural networks like whisper.cpp rely on context and sub-word tokens, making them far better at handling melodic, drawn-out speech without extra training.

and....

For lightweight, fully local word recognition directly in the browser, the best solutions depend on your performance and setup requirements:

## Simpler: @tensorflow-models/speech-commands
This is Google's official TensorFlow.js library explicitly built for offline keyword recognition. It comes pre-trained out-of-the-box with digits 0–9, directional words ("up", "down"), and "yes"/"no".

    Pros: Runs entirely client-side, extremely fast, works on all modern browsers, requires zero model setup for digits.
    GitHub: [tensorflow/tfjs-models/speech-commands](https://github.com/tensorflow/tfjs-models/tree/master/speech-commands)
    Tutorial: Google's official [Audio Recognition Codelab](https://codelabs.developers.google.com/codelabs/tensorflowjs-audio-codelab) shows step-by-step how to load the model and listen for digits.

Quick Example:

```javascript
<script src="https://cdn.jsdelivr.net/npm/@tensorflow/tfjs"></script>
<script src="https://cdn.jsdelivr.net/npm/@tensorflow-models/speech-commands"></script>

<script>
  async function listenForDigits() {
    // Create recognizer (uses the default 18-word dictionary including numbers 0-9)
    const recognizer = speechCommands.create('BROWSER_FFT');
    await recognizer.ensureModelLoaded();

    const labels = recognizer.wordLabels(); // List of words the model recognizes

    // Start listening to mic input
    recognizer.listen(result => {
      // Find highest confidence word
      const maxScoreIndex = result.scores.indexOf(Math.max(...result.scores));
      const detectedWord = labels[maxScoreIndex];
      
      console.log(`Detected: ${detectedWord}`);
    }, { probabilityThreshold: 0.75 });
  }

  listenForDigits();
</script>
```

## For Complex Audio: whisper.cpp (Wasm)

If you need higher accuracy in noisy environments, run OpenAI's Whisper model inside WebAssembly using C++ compilation.

    Pros: High accuracy, completely off-grid.
    Cons: Larger download footprint (~30MB–75MB for the tiny or base model).
    GitHub Wrapper: [Timur00Kh/whisper.wasm](https://github.com/timur00kh/whisper.wasm) (or the core [whisper.cpp WASM example](https://github.com/ggerganov/whisper.cpp/tree/master/examples/whisper.wasm)).

For digits only, TensorFlow.js Speech Commands is far lighter (~1–2MB) and will give you the fastest load times with zero server dependencies.
